code/datums/dash_weapon.dm

```diff
@@ -36,16 +36,30 @@
 	var/turf/T = get_turf(target)
 	if(user in viewers(user.client.view, T))
 		var/obj/spot1 = new phaseout(get_turf(user), user.dir)
-		if(do_teleport(user, T, channel = TELEPORT_CHANNEL_FREE, no_effects = TRUE))
+		var/datum/callback/dashslash_callback = CALLBACK(src, PROC_REF(dashslash), user)
+		var/turf/new_location = do_dash(user, get_turf(user), T, obj_damage=200, phase=FALSE, on_turf_cross=dashslash_callback)
+		if(new_location)
 			playsound(T, dash_sound, 25, 1)
-			var/obj/spot2 = new phasein(get_turf(user), user.dir)
+			var/obj/spot2 = new phasein(new_location, user.dir)
 			spot1.Beam(spot2,beam_effect,time=2 SECONDS)
 			current_charges--
-			owner.update_action_buttons_icon()
+			if (owner)
+				owner.update_action_buttons_icon()
 			addtimer(CALLBACK(src, PROC_REF(charge)), charge_rate)
 		else
 			to_chat(user, "<span class='warning'>You cannot dash here!</span>")
 
+/datum/action/innate/dash/proc/dashslash(mob/user, turf/slash_location)
+	for(var/mob/living/target in slash_location)//Hit everything in the turf
+		// Skip any mobs that aren't standing, or aren't dense
+		if ((target.body_position == LYING_DOWN) || !target.density || user == target)
```

Comment:
If a mob has incorporeal_move at any value then it should not be attacked. incorporeal_move is very poorly coded so isn't checked in a lot of places, but it should be checked here